controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -518,15 +527,16 @@ private long findSegmentSplitsMerges(List<StreamSegmentRecord> referenceSegments
                                 });
     }
 
-    private CompletableFuture<Set<StreamSegmentRecord>> segmentsBetweenStreamCutSpans(Map<StreamSegmentRecord, Integer> spanFrom,
+    @VisibleForTesting
+    CompletableFuture<Set<StreamSegmentRecord>> segmentsBetweenStreamCutSpans(Map<StreamSegmentRecord, Integer> spanFrom,
```

Comment:
Does this method have side-effects? It was private before and I'm worried that we are increasing substantially the visible method surface here for testing purposes.

Reply:
this method only reads and performs computation, doesnt change the metadata state. 
so its visibility (at package level) is safe. 

This class is anyway not instantiated directly and callers deal with `Stream` interface which does not have these apis.